### PR TITLE
Block endpoints

### PR DESCRIPTION
**Pull request by edmundnoble**
I'm not sure if this is exactly the API we want, but I have been able to adapt chainweb-data to use it (https://github.com/kadena-io/chainweb-data/pull/162).

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205272616283861

### CODE REVIEW

File: chainweb.cabal [hunk 1]
```diff
@@ -437,7 +437,7 @@ test-suite chainweb-tests
     default-language: Haskell2010
     ghc-options:
         -threaded
-        -with-rtsopts=-N2
+        -with-rtsopts=-N4
```
Comment (chessai) [inline chainweb.cabal:440]: rationale?

Comment (edmundnoble) [reply chainweb.cabal:440]: This PR takes a lot of sequential test groups and makes them parallel, so we have more opportunities for parallelism between test suites. And I don't really know why it was -N2, it was a bit faster for me using -N4.

File: test/Chainweb/Test/Pact/PactExec.hs [hunk 2]
```diff
@@ -73,8 +73,8 @@ testVersion = slowForkingCpmTestVersion petersonChainGraph
 testEventsVersion :: ChainwebVersion
 testEventsVersion = fastForkingCpmTestVersion singletonChainGraph
 
-tests :: ScheduledTest
```
Comment (chessai) [inline test/Chainweb/Test/Pact/PactExec.hs:76]: bless up

File: test/Chainweb/Test/RestAPI/Client_.hs [hunk 3]
```diff
@@ -163,3 +178,17 @@ branchHeadersClient' v c = runIdentity $ do
...
+    -> Maybe Limit
+    -> Maybe (NextItem BlockHash)
+    -> Maybe MinRank
+    -> Maybe MaxRank
+    -> BranchBounds BlockHeaderDb
+    -> ClientM_ (Page (NextItem BlockHash) Block)
+branchBlocksClient' v c = runIdentity $ do
+    (SomeSing (SChainwebVersion :: Sing v)) <- return $ toSing (_versionName v)
+    (SomeSing (SChainId :: Sing c)) <- return $ toSing c
+    return $ client_ @(BranchBlocksApi v c)
```
Comment (chessai) [inline test/Chainweb/Test/RestAPI/Client_.hs:194]:
```suggestion
    return $ client_ @(BranchBlocksApi v c)
    
```
newline at end of file

File: test/Chainweb/Test/TreeDB/RemoteDB.hs [hunk 4]
```diff
@@ -20,7 +20,7 @@ import Test.Tasty
 tests :: TestTree
 tests = testGroup "RemoteDB"
     [
-    -- , treeDbInvariants withDb Sequential
+    -- , treeDbInvariants withDb (flip sequentialTestGroup AllFinish)
```
Comment (chessai) [inline test/Chainweb/Test/TreeDB/RemoteDB.hs:23]: this seems like it doesn't run anything?

Comment (chessai) [reply test/Chainweb/Test/TreeDB/RemoteDB.hs:23]: fixing this should actually be in a separate PR, nvm.

**Review by chessai — APPROVED** [review]

**Review by ak3n — APPROVED** [review]
LGTM 👍

**Pull request merged**